Authentication/AuthenticationComponentsFactory.swift

```diff
@@ -0,0 +1,353 @@
+//
+//  AuthenticationComponentsFactory.swift
+//  Authentication
+//
+//  Created by Daniela Riesgo on 8/1/16.
+//  Copyright © 2016 Wolox. All rights reserved.
+//
+
+public protocol AuthenticationComponentsFactoryType: LoginComponentsFactory, SignupComponentsFactory {
+    
+    //var sessionService: SessionServiceType { get }
```

Comment:
I wrote in known issue something to discuss.

We should discuss if the session service should go in the ComponentsFactory or not. The issue with having it there is that it will need an associated type because SessionServiceType uses an associated type.